lib/brook.ex

```diff
@@ -225,4 +225,32 @@ defmodule Brook do
       {:error, reason} -> raise reason
     end
   end
+
+  @spec serialize(term) :: {:ok, String.t()} | {:error, term}
+  def serialize(data) do
+    case Application.get_env(:brook, :serializer, :brook) do
```

Comment:
I really want to open this up for extension by passing around modules via configuration. Like this:

```elixir
def serialize(data) do
  s = Application.get_env(:brook, :serializer, Brook.Serializer)
  s.serialize(data)
end
```

The user can choose `Brook.Serializer`, `JsonSerde`, or roll their own and no one has to touch Brook. The problem is that `Brook.Serializer` and `Brook.Deserializer` are two different modules. Some options come to mind:

1. Have `:serializer` and `:deserializer` app configs for Brook. This becomes weird unless `JsonSerde` becomes the default, since I'd have to set both to `JsonSerde`.
2. Add a module/function combo to `:brook_serializer` to allow one module to pass around. Could be `Brook.Serde` and delegate to `Brook.Serializer` and `Brook.Deserializer`? Then you can default accordingly: `Application.get_env(:brook, :serializer, Brook.Serde)`.

Reply:
I like option 2 more, but tell me what you think of the idea in general.

Reply:
I like option 2 as well.

Reply:
I like the second option as well, I will get that done before merging this PR